internal/audit/kafka/tls.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package kafka
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/cerbos/cerbos/internal/observability/logging"
+	"go.uber.org/zap"
+)
+
+const DefaultTLSVersion = tls.VersionTLS12
+
+type TLSReloader struct {
+	cert           *tls.Certificate
+	certPath       string
+	keyPath        string
+	mu             sync.RWMutex
+	reloadInterval time.Duration
+}
+
+func newTLSReloader(ctx context.Context, reloadInterval time.Duration, certPath, keyPath string) (*TLSReloader, error) {
+	reloader := &TLSReloader{
+		certPath:       certPath,
+		keyPath:        keyPath,
+		reloadInterval: reloadInterval,
+	}
+
+	cert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load TLS key pair: %w", err)
+	}
+	reloader.cert = &cert
+
+	go func() {
+		if err := reloader.reload(ctx); err != nil {
+			logging.FromContext(ctx).Named("kafka").Error("Failed to reload TLS certificate", zap.Error(err))
+		}
+	}()
+
+	return reloader, nil
+}
+
+func (r *TLSReloader) reload(ctx context.Context) error {
+	ticker := time.NewTicker(r.reloadInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+
+		case <-ticker.C:
+			cert, err := tls.LoadX509KeyPair(r.certPath, r.keyPath)
+			if err != nil {
+				return fmt.Errorf("failed to load TLS key pair: %w", err)
```

Comment:
I don't think failing to load the files should terminate the reload loop, in case the failure is transient. Otherwise, it'll fail and the only way to recover is to restart the PDP. I reckon we should move the logging in here and keep looping.

Reply:
Good spot - thanks!
https://github.com/cerbos/cerbos/pull/1667/commits/c5975d396505131a787bd7fd733c0b64ae702767